pgjdbc/src/main/java/org/postgresql/jdbc/TimestampUtils.java

```diff
@@ -809,6 +847,61 @@ public Timestamp toTimestampBin(TimeZone tz, byte[] bytes, boolean timestamptz)
     return ts;
   }
 
+  //#if mvn.project.property.postgresql.jdbc.spec >= "JDBC4.2"
+  /**
+   * Returns the SQL Timestamp object matching the given bytes with {@link Oid#TIMESTAMP} or
+   * {@link Oid#TIMESTAMPTZ}.
+   *
+   * @param bytes The binary encoded timestamp value.
+   * @return The parsed timestamp object.
+   * @throws PSQLException If binary format could not be parsed.
+   */
+  public LocalDateTime toLocalDateTimeBin(byte[] bytes)  throws PSQLException {
```

Comment:
Frankly speaking, I'm more inclined to follow Java's Calendar & Co behavior there.
As Calendar parses timestamps, it seems to adjust to the next "valid" date. Even though that is debatable, it is already implemented.

If we really want implement another behavior (e.g. `LocalDateTime` being able to represent `2016-03-27 02:15:00` and `Timestamp` not able), then we need some strong arguments for that.

PS. I do not like how backend returns "2016-03-27 02:15:00 Europe/Berlin" in string format.

PPS. Current implementation does NOT track `PgConnection`'s timezone (that is `set time_zone=...`). I wish it was implemented. It is required for **proper** implementation of `getString()` for timestamp-like data. Can you probably look into that? See here: https://github.com/pgjdbc/pgjdbc/issues/130#issuecomment-170820995